src/Sankey.js

```diff
@@ -0,0 +1,201 @@
+        verticalAlign: "middle"
+      },
+      Path: {
+        fill: "none",
+        label: false,
+        stroke: "#dbdbdb",
+        strokeWidth: 20,
+        strokeOpacity: 0.4
+      }
+    });
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const path = sankeyLinkHorizontal();
+
+    const nodes = this._nodes;
+
+    const nodeLookup = this._nodeLookup = this._nodes.reduce((obj, d, key) => {
+      obj[d[this._nodeId || "id"]] = key;
+      return obj;
+    }, {});
+
+    const links = this._links.map(link => {
+      const check = ["source", "target"];
+
+      const linkLookup = check.reduce((result, item) => {
+        result[item] =
+          typeof link[item] === "number" ? link[item] : nodeLookup[link[item]];
+        return result;
+      }, {});
+
+      return {
+        source: linkLookup.source,
+        target: linkLookup.target,
+        value: link.value
+      };
+    });
+
+    const transform = `translate(${this._margin.left}, ${this._margin.top})`;
+
+    this._sankey
+      .nodeWidth(this._nodeWidth)
+      .nodes(nodes)
+      .links(links)
+      .size([width, height])();
+
+    this._shapes.push(
+      new Path()
+        .config(this._shapeConfig)
+        .config(this._shapeConfig.Path)
+        .data(links)
+        .d(path)
+        .select(
+          elem("g.d3plus-Links", {
+            parent: this._select,
+            enter: {transform},
+            update: {transform}
+          }).node()
+        )
+        .config(assign(this._tooltip, false))
+        .config(
+          configPrep.bind(this)(
+            assign(this._shapeConfig, {
+              Path: {
+                strokeWidth: d =>
+                  Math.max(
+                    1,
+                    Math.abs(d.source.y1 - d.source.y0) *
+                      (d.value / d.source.value) -
+                      20
+                  )
+              }
+            }),
+            "shape",
+            "Path"
+          )
+        )
+        .render()
+    );
+
+    this._shapes.push(
+      new Rect()
+        .config(this._shapeConfig)
+        .config(this._shapeConfig.Path)
+        .data(nodes)
+        .height(d => d.y1 - d.y0)
+        .width(d => d.x1 - d.x0)
+        .x(d => (d.x1 + d.x0) / 2)
+        .y(d => (d.y1 + d.y0) / 2)
+        .select(
+          elem("g.d3plus-sankey-nodes", {
+            parent: this._select,
+            enter: {transform},
+            update: {transform}
+          }).node()
+        )
+        .config(configPrep.bind(this)(this._shapeConfig, "shape", "Rect"))
+        .render()
+    );
+    console.log(this);
+    return this;
+  }
+
+  nodeId(_) {
+    return arguments.length ? (this._nodeId = _, this) : this._nodeId;
+  }
```

Comment:
1. move this method after the `links` method (methods are alpha-sorted)
2. add some [JSDOC](http://usejsdoc.org/) comments!